frontend/src/app/apiSlice.ts

```diff
@@ -0,0 +1,36 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+function getApiUrl(): string {
+  // Avoid an obvious poor environment setup footgun
+  if (process.env.EXPO_PUBLIC_API_DOMAIN == null) {
+    throw new Error("API URL not set");
+  }
+
+  // http in development, https in production
+  if (process.env.NODE_ENV === "development") {
+    return "http://" + process.env.EXPO_PUBLIC_API_DOMAIN;
```

Comment:
lets use the URL object here as well for changing the protocol from http to https

https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol

Reply:
This isn't possible, you have to specify a protocol in the constructor itself for it not to error out.